### PR TITLE
feat: pagination for blog posts

### LINKED ISSUE #643: Add pagination to the posts endpoint
Data from the blog posts endpoint (`/api/v1/posts/`) should be served paginated 
Please use a [Limit & Offset Pagination](https://www.django-rest-framework.org/api-guide/pagination/#limitoffsetpagination) style with the following options:
```
default limit = 3
max limit = 10
```

### LINKED ISSUE #644: Get the blog posts paginated
**Depends on #643**

The blog already displays 3 blog posts at a time, but requests all of them
Make use of the work done in #643 and request a new set of blog posts with each `load more` press

### PR DESCRIPTION
**Pull request by JustBeYou**
### What does it fix?
Implements limit&offset pagination for posts endpoint.

Closes #643 
Closes #644 

### How has it been tested?
Automated test written.

### Note
Should be merged only after #644 is implemented too as the response format has changed and it would break the frontend.

I could do the necessary frontend change tomorrow and merge everything in one PR.

### CODE REVIEW

File: api/blog/views.py [hunk 1]
```diff
@@ -18,13 +18,19 @@ class TagViewSet(viewsets.ModelViewSet):
     serializer_class = TagSerializer
 
 
+class PostsPagination(pagination.LimitOffsetPagination):
+    default_limit = 10
+    max_limit = 50
```
Comment (tudoramariei) [inline api/blog/views.py:23]: the values should be changed to:
```
default_limit = 4
max_limit = 10
```
We don't need to display too many blog posts at a time anywhere so these values should be fine.

**Comment by JustBeYou** [comment]
Fixed. I've also added pagination to the client too.

**Comment by tudoramariei** [comment]
Would you have time to also add pagination to the blog preview? 
It should return only the first 3 posts

**Review by tudoramariei — APPROVED** [review]

**Comment by JustBeYou** [comment]
@tudoramariei done. kinda copy-pasted it, not happy about it, but it works. :stuck_out_tongue_closed_eyes:

**Pull request merged**